test/e2e/e2e_test.go

```diff
@@ -955,6 +955,248 @@ var _ = Describe("e2e multiple external gateway update validation", func() {
+		testContainerFlag := fmt.Sprintf("--container=%s", testContainer)
+		// start the container that will act as an external gateway
+		_, err := runCommand("docker", "run", "-itd", "--privileged", "--network", ciNetworkName, "--name", gwContainerNameAlt1, "centos")
+		if err != nil {
+			framework.Failf("failed to start external gateway test container %s: %v", gwContainerNameAlt1, err)
+		}
+		// retrieve the container ip of the external gateway container
+		exGWIpAlt1, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, gwContainerNameAlt1)
+		if err != nil {
+			framework.Failf("failed to start external gateway test container: %v", err)
+		}
+		// trim newline from the inspect output
+		exGWIpAlt1 = strings.TrimSuffix(exGWIpAlt1, "\n")
+		if ip := net.ParseIP(exGWIpAlt1); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", gwContainerNameAlt1, exGWIpAlt1)
+		}
+		// annotate the test namespace
+		annotateArgs := []string{
+			"annotate",
+			"namespace",
+			f.Namespace.Name,
+			fmt.Sprintf("k8s.ovn.org/routing-external-gws=%s", exGWIpAlt1),
+		}
+		framework.Logf("Annotating the external gateway test namespace to a new container gw:%s ", exGWIpAlt1)
+		framework.RunKubectlOrDie(annotateArgs...)
+		// non-ha ci mode runs a set of kind nodes prefixed with ovn-worker
+		ciWorkerNodeSrc := ovnWorkerNode
+		if haMode {
+			// ha ci mode runs a named set of nodes with a prefix of ovn-control-plane
+			ciWorkerNodeSrc = ovnHaWorkerNode
+		}
+		nodeIP, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("failed to get the node ip address from node %s %v", ciWorkerNodeSrc, err)
+		}
+		nodeIP = strings.TrimSuffix(nodeIP, "\n")
+		if ip := net.ParseIP(nodeIP); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", ciWorkerNodeSrc, nodeIP)
+		}
+		framework.Logf("the pod side node is %s and the source node ip is %s", ciWorkerNodeSrc, nodeIP)
+		podCIDR, err := getNodePodCIDR(ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", ciWorkerNodeSrc, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", ciWorkerNodeSrc, podCIDR)
+		// add loopback interface used to validate all traffic is getting drained through the gateway
+		_, err = runCommand("docker", "exec", gwContainerNameAlt1, "ip", "address", "add", exGWRemoteCidrAlt1, "dev", "lo")
+		if err != nil {
+			framework.Failf("failed to add the loopback ip to dev lo on the test container: %v", err)
+		}
+		// Create the pod that will be used as the source for the connectivity test
+		createGenericPod(f, srcPingPodName, ciWorkerNodeSrc, command)
+
+		// Wait for pod exgw setup to be almost ready
+		wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+			jsonFlag := "jsonpath='{.metadata.annotations.k8s\\.ovn\\.org/routing-external-gws}'"
```

Comment:
the pod wont have this annotation right? Not sure how this doesn't fail?

Reply:
Yeah you're right. I didnt to the polling wait code but after digging into it, there are a few things off here. 
1. It is querying the wrong namespace `ovn-kubernetes` rather than the test framework namespace `f.Namespace.Name` e.g. `multiple-novxlan-externalgw-7887`.
2. The pod is never going to have the `routing-external-gws` annotation. 
3. The kubectl command is missing `-o=` in the `jsonpath=` call. So STDERR is never going to be not null with `error: there is no need to specify a resource type as a separate argument when passing arguments in resource/name form (e.g. 'kubectl get resource/<resource_name>' instead of 'kubectl get resource resource/<resource_name>'`

Example: You could change it to `jsonFlag := "jsonpath='{.metadata.annotations.k8s\\.ovn\\.org/foo}'"` and it will still pass since the annotation isn't going to show up in the pod, its querying the wrong namespace and the command is malformed.

I'm thinking checking for an IP address would work which I do in other functions with [this](https://github.com/ovn-org/ovn-kubernetes/blob/master/test/e2e/e2e_test.go#L867) but using the `wait.PollImmediate` method.

It's basically a `time.Sleep(time.Second * 30)` (which works fine other than not being as efficient in shaving some seconds off the operation). 

If it makes sense to you, should I address that in a separate patch since it's in a few different tests or sort it out in this patch?

Reply:
Thats ok with me. Thanks.